corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -37,23 +61,61 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
 
         ui: {
             submitButton: '#query-submit-button',
+            valueDropdown: 'select.query-field',
         },
 
         events: {
+            'change @ui.valueDropdown': 'changeDropdown',
             'click @ui.submitButton': 'submitAction',
         },
 
-        submitAction: function (e) {
-            e.preventDefault();
-            var payload = {};
-            var fields = $(".query-field");
-            var model = this.parentModel;
-            fields.each(function (index) {
+        getAnswers: function () {
+            var $fields = $(".query-field"),
+                answers = {},
+                model = this.parentModel;
+            $fields.each(function (index) {
                 if (this.value !== '') {
-                    payload[model[index].get('id')] = this.value;
+                    answers[model[index].get('id')] = this.value;
                 }
             });
-            FormplayerFrontend.trigger("menu:query", payload);
+            return answers;
+        },
+
+        changeDropdown: function (e) {
+            e.preventDefault();
+            var self = this;
+            var $fields = $(".query-field");
+
+            // If there aren't at least two dropdowns, there are no dependencies
+            if ($fields.filter("select").length < 2) {
+                return;
+            }
+
+            var Util = hqImport("cloudcare/js/formplayer/utils/util");
+            var urlObject = Util.currentUrlToObject();
+            urlObject.setQuery(this.getAnswers());
+            urlObject.setDoQuery(false);
+            var fetchingPrompts = FormplayerFrontend.getChannel().request("app:select:menus", urlObject);
+            $.when(fetchingPrompts).done(function (response) {
+                for (var i = 0; i < response.models.length; i++) {
+                    var choices = response.models[i].get('itemsetChoices');
+                    if (choices) {
+                        var $field = $($fields.get(i)),
+                            value = parseInt($field.val());
+                        $field.select2('close');    // force close dropdown, the set below can interfere with this when clearing selection
```

Comment:
Is it possible for field to be null? Also, this is the first time I've seen an `$` before a var name, does that just denote that it's an html element?

Reply:
The `$` is a convention denoting that the variable is a jQuery object, which also means it'll never be null. Even if there aren't any matched elements (not an issue here, but would be with something like `$field = $(".something")`), you can call functions on it and they'll just do nothing.